tests/contrib/test_sqlalchemy.py

```diff
@@ -0,0 +1,125 @@
+    credentials = Column(CredentialsType)
+    flow = Column(FlowType)
+
+
+class TestSQLAlchemyStorage(unittest.TestCase):
+    engine = create_engine('sqlite://')
+
+    @classmethod
+    def setUpClass(cls):
+        Base.metadata.create_all(cls.engine)
+        cls.session = sessionmaker(bind=cls.engine)
+
+    def setUp(self):
+        self.credentials = OAuth2Credentials(
+            access_token='token',
+            client_id='client_id',
+            client_secret='client_secret',
+            refresh_token='refresh_token',
+            token_expiry=datetime.datetime.utcnow(),
+            token_uri=GOOGLE_TOKEN_URI,
+            user_agent='DummyAgent',
+        )
+
+    def tearDown(self):
+        session = self.session()
+        session.query(DummyModel).filter_by(id=1).delete()
+        session.commit()
+
+    def compare_credentials(self, result):
+        self.assertEqual(result.access_token, self.credentials.access_token)
+        self.assertEqual(result.client_id, self.credentials.client_id)
+        self.assertEqual(result.client_secret, self.credentials.client_secret)
+        self.assertEqual(result.refresh_token, self.credentials.refresh_token)
+        self.assertEqual(result.token_expiry, self.credentials.token_expiry)
+        self.assertEqual(result.token_uri, self.credentials.token_uri)
+        self.assertEqual(result.user_agent, self.credentials.user_agent)
+
+    def test_locked_get(self):
+        session = self.session()
+        session.add(DummyModel(
+            key=1,
+            credentials=self.credentials,
+        ))
+        session.commit()
+
+        ret = Storage(
+            session=session,
+            model_class=DummyModel,
+            key_name='key',
+            key_value=1,
+            property_name='credentials',
+        ).locked_get()
+
+        self.compare_credentials(ret)
+
+    def test_locked_put(self):
+        session = self.session()
+        Storage(
+            session=session,
+            model_class=DummyModel,
+            key_name='key',
+            key_value=1,
+            property_name='credentials',
+        ).locked_put(self.credentials)
+        session.commit()
+
+        ret = session.query(DummyModel).filter_by(key=1).first()
+        self.compare_credentials(ret.credentials)
+
+    def test_locked_delete(self):
+        session = self.session()
+        session.add(DummyModel(
+            key=1,
+            credentials=self.credentials,
+        ))
+        session.commit()
+
+        q = session.query(DummyModel).filter_by(key=1)
+        self.assertTrue(q.first() is not None)
+        Storage(
+            session=session,
+            model_class=DummyModel,
+            key_name='key',
+            key_value=1,
+            property_name='credentials',
+        ).locked_delete()
+        session.commit()
+        self.assertTrue(q.first() is None)
+
+
+if __name__ == '__main__':  # pragma: NO COVER
```

Comment:
Sorry, I was using this to run tests myself. And again, it was copied from existing module.